Content.Shared/Automod/AutomodActionGroupPrototype.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System.Runtime.CompilerServices;
+using System.Text.RegularExpressions;
+using Content.Shared.Database;
+using JetBrains.Annotations;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+
+namespace Content.Shared.Automod;
+
+/// <summary>
+/// An automod filter for running actions when a pattern match is found in user input.
+/// </summary>
+/// <param name="pattern">The text to find in user input.</param>
+/// <param name="filterType">The type of filter to sort this definition into.</param>
+/// <param name="actionGroup">The <seealso cref="AutomodActionGroupPrototype"/> to run when the <paramref name="pattern"/> is matched</param>
+/// <param name="targets">The user input types that this filter applies to.</param>
+/// <param name="name">The user facing name of this filter.</param>
+[Virtual]
+public class AutomodFilterDef(
+    int? id,
+    string pattern,
+    AutomodFilterType filterType,
+    ProtoId<AutomodActionGroupPrototype> actionGroup,
+    AutomodTarget targets,
+    string name)
+{
+    public readonly int? Id = id;
+    public readonly string Pattern = pattern;
+    public readonly AutomodFilterType FilterType = filterType;
+    public readonly string ActionGroup = actionGroup;
+    public readonly AutomodTarget TargetFlags = targets;
+    public readonly string DisplayName = name;
+
+    public AutomodFilterDef(
+        string pattern,
+        AutomodFilterType filterType,
+        string actionGroup,
+        AutomodTarget targets,
+        string name) : this(null, pattern, filterType, actionGroup, targets, name)
+    {
+    }
+}
+
+/// <summary>
+/// <see cref="AutomodFilterDef"/> with Regex saved.
+/// </summary>
+/// <inheritdoc cref="AutomodFilterDef"/>
+/// <param name="regex">The compiled Regex from the <paramref name="pattern"/>.</param>
+public sealed class RegexAutomodFilterDef(
```

Comment:
Do not use inheritance to just add more data to a type. This type should just contain the original `AutomodFilterDef` instance directly.

Reply:
Fixed in [ea7a88b](https://github.com/space-wizards/space-station-14/pull/28570/commits/ea7a88b94442619baacb0fb282c02888e0fad884)